lib/validator/validator.go

```diff
@@ -286,16 +285,28 @@ func validateHostEndpointSpec(v *validator.Validate, structLevel *validator.Stru
 	}
 }
 
-func validatePoolMetadata(v *validator.Validate, structLevel *validator.StructLevel) {
-	pool := structLevel.CurrentStruct.Interface().(api.IPPoolMetadata)
+func validatePool(v *validator.Validate, structLevel *validator.StructLevel) {
+	pool := structLevel.CurrentStruct.Interface().(api.IPPool)
+
+	// Metadata values may be nil when doing a list operation, so skip
+	// validation if not assigned.
+	if pool.Metadata.CIDR.IP == nil {
+		return
+	}
+
+	// IPIP cannot be enabled for IPv6.
+	if pool.Metadata.CIDR.Version() == 6 && pool.Spec.IPIP != nil && pool.Spec.IPIP.Enabled {
+		structLevel.ReportError(reflect.ValueOf(pool.Spec.IPIP.Enabled),
+			"IPIP.Enabled", "", reason("IPIP is not supported on an IPv6 IP pool"))
+	}
 
-	// The Calico IPAM places restrictions on the minimum IP pool size, check that the
-	// pool is at least the minimum size.
-	if pool.CIDR.IP != nil {
-		ones, bits := pool.CIDR.Mask.Size()
-		log.Debugf("Pool CIDR: %s, num bits: %d", pool.CIDR, bits-ones)
+	// The Calico IPAM places restrictions on the minimum IP pool size.  If
+	// the pool is enabled, check that the pool is at least the minimum size.
+	if !pool.Spec.Disabled {
```

Comment:
I wish the PoolMetadata check could have remained separate but I see that this is also changing the behavior to only apply the CIDR size check when the pool is Enabled.
In the previous check, `CIDR.IP` was checked for not null before checking the CIDR Mask, I don't understand the reasoning for that, but should that be done here too?